music21/stream/tests.py

```diff
@@ -2501,6 +2501,34 @@ def testMakeAccidentalsTies(self):
                              '%s failed, %s != %s' %
                                 (i, allNotes[i].pitch.accidental.displayStatus, ds[i]))
 
+    def testMakeNotationTiesKeyless(self):
+        from music21 import converter
+        p = converter.parse('tinynotation: 4/4 f#1~ f#1')
+        # Key of no sharps/flats
+        p.measure(1).insert(0, key.KeySignature(sharps=0))
+        # makeNotation performs some checks before calling makeAccidentals
+        p.makeNotation(inPlace=True)
+        self.assertEqual(p.measure(2).notes.first().pitch.accidental.displayStatus, False)
+
+    def testMakeNotationTiesKeyChange(self):
+        from music21 import converter
+        p = converter.parse('tinynotation: 4/4 f#1~ f#1')
+        # Insert key change where held-over note is chromatic
+        p.measure(2).insert(0, key.KeySignature(sharps=-1))
+        pMade1 = p.makeNotation()
+        self.assertEqual(pMade1.measure(2).notes.first().pitch.accidental.displayStatus, False)
```

Comment:
I couldn't find anything about this case directly in the Gould, _Behind Bars_ book, but the closest recommendation is if there's a system break mid-tie to repeat the accidental afterwards (she says at one point, better without parentheses, but then demonstrates it with parentheses more often.  We don't have a displayParentheses status unfortunately).

Reply:
Ah, but after the tie note concludes, yes? Not on the terminus of the tie itself? I've never seen an accidental interrupt a tie: https://github.com/cuthbertLab/music21/issues/865#issuecomment-782503317

Reply:
But I will zip over to IMSLP and pull down some Reger to find out ...

Reply:
This edition of a Reger intermezzo validates Gould's point about system breaks employing a parenthetical accidental on a chromatic end-tie:

![Screen Shot 2021-02-23 at 2 18 53 PM](https://user-images.githubusercontent.com/38668450/108895934-3c520900-75e2-11eb-9001-424a38ae8af2.png)
![Screen Shot 2021-02-23 at 2 18 58 PM](https://user-images.githubusercontent.com/38668450/108895944-3eb46300-75e2-11eb-8e6f-8bdceb32420c.png)

Reply:
Ugh, putting the key change AFTER the tie ends is just cheating, robbing us of a model:
![Screen Shot 2021-02-23 at 2 23 07 PM](https://user-images.githubusercontent.com/38668450/108896306-bc786e80-75e2-11eb-8940-8cb32dc1681a.png)

Reply:
Too cumbersome to post a photo of a hard copy, but now I've got an example 12 measures from the end of Messiaen, op.1/vi, "Cloches d'angoisse et larmes d'adieu" where the accidentals are reiterated on the end-tie across a key change (and for one of the three piano staves, a page break also) on a note chromatic in the new key.

B major chord being tied over during a key change from 5 sharps to 2 sharps. The D# was explicit on the start tie because of an earlier D-natural. Then the D# is reiterated after the barline and key change (the key change was printed with cancelled naturals after the two sharps)

Reply:
I arranged that prelude in college, and I had a copy sticking out of my bag when an acquaintance spotted it, and assuming it was an original composition, expressed alarm at what seemed to be a pretentious title, translating on the fly, "Bells of ...sadness and tears of ... goodbye?!" as if to ask, "are you doing alright, fella?"

Reply:
`tiePitchSet` should have chromatic notes (in the new key) filtered out if `m.keySignature` is not None, then, so they get reiterated, maybe.